src/python/pants/engine/legacy/parser.py

```diff
@@ -140,27 +142,50 @@ def parse(self, filepath: str, filecontent: bytes):
     # Note that this is incredibly poor sandboxing. There are many ways to get around it.
     # But it's sufficient to tell most users who aren't being actively malicious that they're doing
     # something wrong, and it has a low performance overhead.
-    if self._build_file_imports_behavior != 'allow' and 'import' in python:
+    if "globs" in python or (self._build_file_imports_behavior != 'allow' and 'import' in python):
       io_wrapped_python = StringIO(python)
       for token in tokenize.generate_tokens(io_wrapped_python.readline):
-        if token[1] == 'import':
-          line_being_tokenized = token[4]
-          if self._build_file_imports_behavior == 'warn':
-            logger.warning('{} tried to import - import statements should be avoided ({})'.format(
-              filepath,
-              line_being_tokenized
-            ))
-          elif self._build_file_imports_behavior == 'error':
-            raise ParseError(
-              'import statements have been banned, but tried to import: {}'.format(
-                line_being_tokenized
-              )
+        token_str = token[1]
+        lineno, _ = token[2]
+
+        # We have this deprecation here, rather than in `engine/legacy/structs.py` where the
```

Comment:
So... it feels like this could possibly be more generally implemented as "setting the filename that we are `eval`ing", which would cause deprecations implemented in our standard locations to get blamed back to a file via a python stack trace?

Reply:
I don't follow. Do you mean "This works, but we could make a broader change to start passing more information down the stack trace so that future error messages could be more helpful?"

 If so, I think I agree but would prefer punting on that until we have an actual use case for it.

Reply:
Sorry. Concretely what I meant here was to replace:
```
exec(python, dict(self._symbols))
```
with:
```
compiled = compile(python, filepath, 'exec')
exec(compiled, dict(self._symbols))
```

I _think_ that this would have the effect of causing our existing deprecation warning infrastructure to be triggered "from" the `BUILD` filename.

Example from a repl:
```
>>> compiled = compile('raise Exception("blah")', 'an/example/filename.py', 'exec')
>>> exec(compiled)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "an/example/filename.py", line 1, in <module>
Exception: blah
>>>
```

Reply:
Ah, that would be neat! Unfortunately doesn't seem to make a difference. I also tried playing with `stacklevel` and couldn't get `[WARN] /Users/eric/DocsLocal/code/projects/pants/src/python/pants/engine/native.py:486:` to point to the BUILD file, only to different files like `build_files.py` and `mapper.py`. 

I experimented with `context=0` but we print `'<no code context available>'` in that case, which doesn't seem super helpful. Maybe in a followup we should change `context=0` to mean don't print any lines of context -- only the deprecation warning? Generally, `context=0` seems like a good default - I'm not sure why we print the line causing the deprecation in the first place as that seems like a leaky abstraction.

Reply:
Blargh. Ok.

Is it possible to break out the deprecation-specific portion of this into a separate method so it's clearer what to delete as part of the removal?